### DISCUSSION TITLE
How to get information of superscript and subscript text in the pdf

### DISCUSSION THREAD

**Discussion by sarveshkrg** [jsvine/pdfplumber#730] [Ideas]
Hi, anyone please help to get superscript subscript text identification from normal text in the pdf so that I can save the chemical formula and other constant values present in the pdf as it is into the excel. Thanks in advance.
Input
![Capture](https://user-images.githubusercontent.com/46168717/189921898-797fd185-cb6d-4afd-a106-c6cb967128f1.PNG)

Output
![Capture2](https://user-images.githubusercontent.com/46168717/189921905-f9640a96-5530-494b-8534-a7cbf28baf08.PNG)

**Comment by jsvine** [comment]
Hi @sarveshkrg, and thanks for your interest in this library. Subscript and superscript do not have their own concepts in PDFs; rather, they're just text that use smaller font sizes and positioning. I recommend examining the `size` and `bottom`/`top` attributes of the objects in `Page.chars`. From there, you may have to write some custom code to extract the characters in superscript in a way that suits your particular goals.

**Comment by gulfamhussain80** [comment]
@jsvine I am trying to follow this approach, but the results are inconsistent—sometimes accurate, but other times superscripts are extracted as subscripts and vice versa. 
@sarveshkrg  Were you able to find a solution, I am also facing a similar problm.

**Reply by petermr** [reply]
The relative positions of superscripts (and subscripts) are dependent on the font and the publishing tool that the authors use. Note how the `-23` is differently placed in the san-serif and serif fonts. So your code has to use both the sizes and relative positions. 
The superscript/subscript problem can occur when the lines are close together and it's not clear to which line the sub/superscript applies. If it's a single line then it looks like an error of logic.
